srtcore/utilities.h

```diff
@@ -77,6 +77,21 @@ written by
 // Windows warning disabler
 #define _CRT_SECURE_NO_WARNINGS 1
 
+#if defined(__linux__) || defined(__CYGWIN__)
+   // These macros needed to be defined for some versions of GLIBC to ensure that
+   // the required macros are defined in system headers included later in this
+   // file.
+   #ifndef _BSD_SOURCE
+      #define _BSD_SOURCE
+   #endif
+   #ifndef __USE_BSD
+      #define __USE_BSD
+   #endif
+   #ifndef _DEFAULT_SOURCE
+      #define _DEFAULT_SOURCE
+   #endif
+#endif
+
```

Comment:
If this is required only in some versions of GLIBC, then please make this conditional for those versions of GLIBC.

Reply:
@ethouris After more testing. That part of the fix is not really needed.

Reply:
That's what I like more :)